### PR TITLE
App op (atleast_1d and atleast_2d and atleast_3d) | feat (torchlib)

### PR DESCRIPTION
**Pull request by titaiwangms**
op.SequenceMap comes to the rescue for element iteration cases. (Maybe it's just me that didn't know this...)
This PR helps complete #762

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -666,22 +666,54 @@
...
 
-    raise NotImplementedError()
+    @graph()
+    def reshape_to_2d(tensor):
+        shape = op.Shape(tensor)
+        rank = op.Size(shape)
+        if rank == 0:
+            tensor = op.Reshape(tensor, op.Constant(value_ints=[1, 1]))
+        elif rank == 1:
+            tensor = op.Unsqueeze(tensor, op.Constant(value_ints=[0]))
```
[the remark this answers is not in the document]
Comment (gramalingam) [reply onnxscript/function_libs/torch_lib/ops/core.py:695]: `return tensor` is missing

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 2]
```diff
@@ -666,22 +666,54 @@ def aten_atanh(self: TFloat) -> TFloat:
...
+@torch_op("aten::atleast_2d")
+def aten_atleast_2d(self: Sequence[TTensor]) -> TTensor:
     """atleast_2d(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    @graph()
+    def reshape_to_2d(tensor):
+        shape = op.Shape(tensor)
+        rank = op.Size(shape)
+        if rank == 0:
```
[the remark this answers is not in the document]
Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/core.py]: Done

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/core.py]: It's weird the test still pass. I need to revisit the test..

File: onnxscript/function_libs/torch_lib/ops/core.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (titaiwangms) [reply onnxscript/function_libs/torch_lib/ops/core.py]: Also applied on 3d

**Review by justinchuby — APPROVED** [review]
Nice! I didn’t know about this either

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 3]
```diff
@@ -808,6 +814,21 @@ def _where_input_wrangler(
         matcher=lambda sample: len(sample.args) != 2,
         reason="arange_start_step overload takes three arguments (input, start, step)",
     ),
+    skip(
+        "atleast_1d_single_tensor",
+        matcher=lambda sample: isinstance(sample.input, (list, tuple)),
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:819]: So we don’t take a Sequence as input? Where is this op used? How are the inputs produced?

Comment (titaiwangms) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py:819]: It accepts both tensor and list.

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 4]
```diff
@@ -666,22 +666,86 @@ def aten_atanh(self: TFloat) -> TFloat:
...
+def aten_atleast_3d(self: Sequence[TTensor]) -> TTensor:
     """atleast_3d(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    @graph()
+    def reshape_to_3d(tensor):
+        shape = op.Shape(tensor)
+        rank = op.Size(shape)
+        if rank <= 1:
+            tensor = op.Reshape(tensor, op.Constant(value_ints=[1, -1, 1]))
```
Comment (gramalingam) [inline onnxscript/function_libs/torch_lib/ops/core.py:730]: You may want to double-check the downstream usage to make sure whether you want to reshape a tensor of size N to 1x1xN or Nx1x1 ... the above reshapes it to 1xNx1 .... which may be okay, but just wondering.

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py (hunk 3 above)
Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py:819]: Oh I see there are two variants

**Pull request merged**